test/e2e-tests.sh

```diff
@@ -40,6 +40,59 @@ function dump_extra_cluster_state() {
   kubectl -n knative-build-pipeline logs $(get_app_pod build-pipeline-webhook knative-build-pipeline)
 }
 
+function run_yaml_tests() {
+  echo ">> Starting tests"
+
+  sed -i.bak 's/christiewilson-catfactory/${KO_DOCKER_REPO}/' examples/resources.yaml
+
+  ko apply -R -f examples/ || return 1
+
+  # Wait for tests to finish.
+  echo ">> Waiting for tests to finish"
+  local tests_finished=0
+  for i in {1..60}; do
+    local finished="$(kubectl get pipelinerun.pipeline.knative.dev --output=jsonpath='{.items[*].status.conditions[*].status}')"
+    if [[ ! "$finished" == *"Unknown"* ]]; then
+      tests_finished=1
+      break
+    fi
+    sleep 5
+  done
+  if (( ! tests_finished )); then
+    echo "ERROR: tests timed out"
+    return 1
+  fi
+
+  # Check that tests passed.
+  local failed=0
+  echo ">> Checking test results"
+  for expected_status in succeeded failed; do
+    results="$(kubectl get pipelinerun.pipeline.knative.dev -l expect=${expected_status} \
+	--output=jsonpath='{range .items[*]}{.metadata.name}={.status.conditions[*].type}{.status.conditions[*].status}{" "}{end}')"
+    case $expected_status in
+      succeeded)
+	want=succeededtrue
+	;;
+      failed)
+	want=succeededfalse
+	;;
+      *)
+	echo "ERROR: Invalid expected status '${expected_status}'"
+	failed=1
+	;;
+    esac
+    for result in ${results}; do
+      if [[ ! "${result,,}" == *"=${want}" ]]; then
+	echo "ERROR: test ${result} but should be ${want}"
+	failed=1
+      fi
+    done
+  done
+  (( failed )) && return 1
+  echo ">> All YAML tests passed"
+  return 0
+}
```

Comment:
could the commit message maybe include the fact that this came from https://github.com/knative/build/blob/f7397a3ec59278416aac4890a5a08783c86e5353/test/e2e-common.sh , just in case we need to track it down one day?

(p.s. would be awesome to re-implement this in Go or even Python - but np if you dont want to do that now)

Reply:
yeah, I wanted to get it done in bash first as the fastest way, then refactor later. Probably Go is better so we don't introduce another language